transformer_engine/pytorch/ops/basic/dropout.py

```diff
@@ -38,21 +40,39 @@ def op_forward(
     ) -> torch.Tensor:
 
         # Compute dropout if training
-        out = input_
         is_training = self.training
+        out = input_
         mask = None
         if is_training:
-            keep_prob = 1 - self.dropout_probability
-            mask = torch.empty_like(input_)
-            mask.bernoulli_(keep_prob)
-            mask *= 1 / keep_prob
-            out = out * mask
-
+            if input_.numel() % 16 != 0:
+                assert not isinstance(
+                    input_, Float8TensorBase
+                ), "fp8 dropout does not support non-16-element aligned inputs"
```

Comment:
@timmoon10 Should we fail here or dequantize instead?

Reply:
That would be a good approach. If we get unsupported data, e.g. FP8 with bad dims or MXFP8, then we can just cast to a plain PyTorch tensor and apply the high-precision kernel.